lib/oauth2/access_token.rb

```diff
@@ -32,22 +32,25 @@ def from_kvform(client, kvform)
     # @option opts [String] :refresh_token (nil) the refresh_token value
     # @option opts [FixNum, String] :expires_in (nil) the number of seconds in which the AccessToken will expire
     # @option opts [FixNum, String] :expires_at (nil) the epoch time in seconds in which AccessToken will expire
+    # @option opts [FixNum, String] :expires_latency (nil) the number of seconds by which AccessToken validity will be reduced to offset latency
     # @option opts [Symbol] :mode (:header) the transmission mode of the Access Token parameter value
     #    one of :header, :body or :query
     # @option opts [String] :header_format ('Bearer %s') the string format to use for the Authorization header
     # @option opts [String] :param_name ('access_token') the parameter name to use for transmission of the
     #    Access Token value in :body or :query transmission mode
-    def initialize(client, token, opts = {}) # rubocop:disable Metrics/AbcSize
+    def initialize(client, token, opts = {}) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
       @client = client
       @token = token.to_s
       opts = opts.dup
-      [:refresh_token, :expires_in, :expires_at].each do |arg|
+      [:refresh_token, :expires_in, :expires_at, :expires_latency].each do |arg|
         instance_variable_set("@#{arg}", opts.delete(arg) || opts.delete(arg.to_s))
       end
+      @expires_latency ||= 0
       @expires_in ||= opts.delete('expires')
       @expires_in &&= @expires_in.to_i
       @expires_at &&= @expires_at.to_i
-      @expires_at ||= Time.now.to_i + @expires_in if @expires_in
+      @expires_latency &&= @expires_latency.to_i
+      @expires_at ||= Time.now.to_i + @expires_in - @expires_latency if @expires_in
```

Comment:
This is almost correct.
The instance variables initially get set to a value if provided, or `nil`.
```
instance_variable_set("@#{arg}", opts.delete(arg) || opts.delete(arg.to_s))
```
The `&&=` then ensures that the values *stay* `nil`, or if they are String/Integer, they are cast to integer.
```
@expires_in &&= @expires_in.to_i
@expires_at &&= @expires_at.to_i
@expires_latency &&= @expires_latency.to_i
```
This means that any of the values can be `nil`, and if nil, can't be used in math, unless cast to integer, or protected by a condition.
```
@expires_at ||= Time.now.to_i + @expires_in if @expires_in
@expires_at -= @expires_latency if @expires_latency
```
The other difference between this and what you have, is for this version the `@expires_latency` is always applied, if set, even if an exact `:expires_at` time was provided.  I think that this is in  line with the underlying purpose of `@expires_latency` though.